src/Tools/ExternalAccess/OmniSharp/Rename/OmniSharpRenamer.cs

```diff
@@ -12,13 +12,20 @@ namespace Microsoft.CodeAnalysis.ExternalAccess.OmniSharp
 {
     internal static class OmniSharpRenamer
     {
-        public static Task<ConflictResolution> RenameSymbolAsync(
+        public static async Task<Solution> RenameSymbolAsync(
             Solution solution,
             ISymbol symbol,
             string newName,
             OmniSharpRenameOptions options,
             ImmutableHashSet<ISymbol>? nonConflictSymbols,
             CancellationToken cancellationToken)
-            => Renamer.RenameSymbolAsync(solution, symbol, newName, options.ToRenameOptions(), nonConflictSymbols, cancellationToken);
+        {
+            var resolution = await Renamer.RenameSymbolAsync(solution, symbol, newName, options.ToRenameOptions(), nonConflictSymbols, cancellationToken).ConfigureAwait(false);
+
+            if (resolution.ErrorMessage != null)
+                throw new ArgumentException(resolution.ErrorMessage);
```

Comment:
How is Omnisharp expected to handle this? Asking them to catch all ArgumentExceptions and then display that to the user seems unwise if anything in here internally throws something else. Create a custom exception type for them, or maybe return a tuple of `(Solution?, string errorMessage)`?

Reply:
It's already handling it. I have not changed this.

Reply:
https://github.com/OmniSharp/omnisharp-roslyn/blob/ff21ac475cdf8afb15579f2b5fb346945db2a7e9/src/OmniSharp.Roslyn.CSharp/Services/Refactoring/RenameService.cs#L48-L51

Reply:
You're right, using ArgumentException is not ideal.